src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,200 @@
+import React, {Component} from 'react';
+import Hark from 'hark';
+import PropTypes from 'prop-types';
+import Graphicequaliser from './canvasequalizer/canvasEqualizer';
+import PlayButton from './playbutton/playButton';
+import Uploadbutton from './upload/uploadButton';
+import Infoabouttrack from './upload/infoaboutfile/infoAboutUploadFile';
+import Streambutton from './streambutton/streamButton';
+import {connect} from 'react-redux';
+import {createAudioData, playPauseSoundFromFile, createStreamData, startMuteStreamAudio, mergeCanvasWidth} from '../../actions/audioActions';
+
+class Equalizer extends Component {
+  state = {
+    analyser: this.props.audioData.analyser,
+    ctx: undefined,
+    numPoints: undefined,
+    heightArray: undefined
+  }
+
+  componentDidMount() {
+    let canvasEl = document.querySelector("canvas").getContext("2d");
+    let numPoints = this.state.analyser.frequencyBinCount - 80;
+    let uint8Array = new Uint8Array(numPoints);
+    this.setState({
+      ctx: canvasEl,
+      numPoints: numPoints,
+      heightArray: uint8Array
+    });
+    this.detectStreamSoundFromMicrophon();
```

Comment:
i would suggest also to rename this function because of misspelling i guess here should be Microphone

Reply:
renamed